tests/edge_range.cpp

```diff
@@ -0,0 +1,108 @@
+#include "kagen/edge_range.h"
+
+#include "kagen/kagen.h"
+
+#include <gtest/gtest.h>
+
+#include "tests/gather.h"
+#include "tests/utils.h"
+#include "tools/converter.h"
+#include "tools/geometry.h"
+
+using namespace kagen;
+
+void check_edge_range(const Graph& graph) {
+    Edgelist edgelist = graph.edges;
+    if (graph.representation == GraphRepresentation::CSR) {
+        edgelist = BuildEdgeListFromCSR(graph.vertex_range, graph.xadj, graph.adjncy);
+    }
+    EdgeRange edge_range(graph);
+
+    {
+        std::size_t expected_index = 0;
+        for (auto it = edge_range.begin(); it != edge_range.end(); ++it) {
+            auto edge = *it;
+            EXPECT_EQ(it.edge_index(), expected_index);
+            EXPECT_EQ(*it, edge);
+            ++expected_index;
+        }
+    }
+
+    {
+        EXPECT_EQ(edge_range.size(), edgelist.size());
+        for (std::size_t i = 0; auto elem: edge_range) {
+            EXPECT_EQ(elem, edgelist[i]);
+            ++i;
+        }
+    }
+}
```

Comment:
consider putting this directly in the assertion via matchers, this makes the test output more readable.